src/config.rs

```diff
@@ -161,6 +161,9 @@ pub struct Config {
 
     // CLI args passed to ztunnel at runtime
     pub proxy_args: String,
+
+    // custom DNS nameservers; primarily for testing
+    pub dns_nameservers: Vec<SocketAddr>,
```

Comment:
Should this be an option? If set, even if it's empty, I think the expectation would be that it overrides the system settings.